pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -626,20 +626,25 @@ struct ContractionOpConversion : public OpConversionPattern<ContractionOp> {
     fillBuffer(rewriter, loc, cionAdaptor.init(), alloc.resultMemRef,
                alloc.rankedTensorType.getShape());
 
-    // Determine ranges
-    SmallVector<int64_t, 8> ranges;
+    // Determine lower and upper bounds.
+    SmallVector<AffineExpr, 8> ubExpr;
     auto lowerBounds = op.lowerBounds().getValue();
     auto upperBounds = op.upperBounds().getValue();
     assert(lowerBounds.getNumResults() == upperBounds.getNumResults() &&
            "mismatched dims for lower and upper bounds");
     for (unsigned i = 0; i < lowerBounds.getNumResults(); i++) {
-      auto rangeExpr = upperBounds.getResult(i) - lowerBounds.getResult(i) + 1;
-      auto range = rangeExpr.cast<AffineConstantExpr>().getValue();
-      ranges.emplace_back(range);
+      auto uExpr = upperBounds.getResult(i) + 1;
+      auto upper = uExpr.cast<AffineConstantExpr>().getValue();
+      ubExpr.push_back(rewriter.getAffineConstantExpr(upper));
     }
 
+    auto ubMap =
+        AffineMap::get(0, 0, (ArrayRef<AffineExpr>)ubExpr, op.getContext());
```

Comment:
Use `{ubExpr}` instead of using a c-style cast.

Reply:
Actually, why do we need a cast at all? Won't `ubExprs` work here?

Reply:
Casting was needed because the compiler got a error with unable to deduce type. Using the C++ style helped the compiler along and cast is not needed.